backend/src/main/java/de/neuefische/smartcount/SmartCountService.java

```diff
@@ -12,30 +12,40 @@
 
     private final ExpensesRepository expensesRepository;
 
-        public Expense createExpense(Expense expense) {
-           return expensesRepository.save(expense);
+    public Expense createExpense(Expense expense) {
+        return expensesRepository.save(expense);
+    }
+
+    public Collection<Expense> getExpenses() {
+        return expensesRepository.findAll()
+                .stream()
+                .toList();
+    }
+
+    private double getSum() {
+        return expensesRepository.findAll()
+                .stream()
+                .mapToDouble(a -> a.getAmount()).sum();
+    }
+
+    public ExpenseDTO getExpensesDTO() {
+        return new ExpenseDTO(getExpenses(), getSum());
+    }
+
+    public void deleteExpense(String id) {
+        expensesRepository.deleteById(id);
+    }
+
+    public Expense editExpense(String id, Expense expense) {
```

Comment:
```suggestion
     return expensesRepository.findById(id)
             .map(expense -> expenseReoository.save(expense);
```
That way you can use the `ResponseEntity.of`-method inside the controller. It returns a 404 if the `Optional` is empty and 200 if not.